eventbrite/app/controllers.js

```diff
@@ -1,74 +1,68 @@
-type Ticket = {|
-  eventbriteId: string,
-  onSaleStatus: OnSaleStatus,
-  ticketType: TicketType,
-  saleStarts: string, // This isn't a date, as we want it in the display format
-  cost: ?number // in pence
-|};
-
-async function getEventbriteEventTickets(id: string) {
-  const eventTickets = await superagent.get(`${eventbriteApiRoot}/events/${id}/ticket_classes/?token=${eventbritePersonalOauthToken}`);
+export async function getEventsInfo(ctx, next) {
+  // expects ids to come in as id:boolean,id:boolean,id:boolean
+  const ids = ctx.query.ids || '';
+  const eventIds = ids.split(',');
 
-  const tickets = eventTickets.body.ticket_classes.map(ticketClass => {
-    const onSaleStatus =
-      ticketClass.on_sale_status === 'AVAILABLE' ? 'available'
-        : ticketClass.on_sale_status === 'SOLD_OUT' ? 'sold_out'
-          : ticketClass.on_sale_status === 'NOT_YET_ON_SALE' ? 'not_yet_on_sale' : 'unavailable';
-    const ticketClassName = ticketClass.name.toLowerCase();
-    const ticketType: TicketType =
-      ticketClassName.startsWith('waiting list') ? 'waitinglist'
-        : ticketClassName.startsWith('comp ') ? 'comp' : 'standard';
-    const saleStarts = formatTimeDate(new Date(ticketClass.sales_start));
-    const cost = ticketClass.cost && ticketClass.cost.value;
-    return ({onSaleStatus, ticketType, saleStarts, eventbriteId: ticketClass.event_id, cost}: Ticket);
-  }).filter(ticket => ticket.ticketType === 'standard'); // we only want to show standard tickets for now...
+  const eventsRequestPromises = eventIds.map(async (id) => {
+    const eventResponse = await superagent.get(`${eventbriteApiRoot}/events/${id}/?token=${eventbritePersonalOauthToken}`);
+    const event = eventResponse.body;
 
-  return tickets;
-}
+    if (event.is_series === false) {
+      const eventTickets = await superagent.get(`${eventbriteApiRoot}/events/${id}/ticket_classes/?token=${eventbritePersonalOauthToken}`);
 
-export async function renderEventbriteTicketStatus(ctx, next) {
-  const id = ctx.params.id;
-  const tickets = await getEventbriteEventTickets(id);
+      const tickets = eventTickets.body.ticket_classes.filter(ticketClass => {
+        const ticketClassName = ticketClass.name.toLowerCase();
+        const ignore = ticketClassName.startsWith('waiting list') ||
+          ticketClassName.startsWith('comp');
+        return !ignore;
+      }).map(ticketClass => {
+        const isFullyBooked = ticketClass.on_sale_status === 'SOLD_OUT';
+        const bookingOpens = new Date(ticketClass.sales_start);
+        const salesEnded = new Date(ticketClass.sales_end) < new Date();
 
-  const standardTicket = tickets.find(ticket => ticket.ticketType === 'standard');
-  if (standardTicket) {
-    ctx.render('components/eventbrite-button/eventbrite-ticket-status', {
-      ticket: standardTicket
-    });
+        return {
+          isFullyBooked,
+          bookingOpens,
+          salesEnded
+        };
+      });
+      return tickets.find(_ => _);
```

Comment:
Does this just return the first? Wouldn't `tickets[0]` be better if that's the intention?